### ISSUE TITLE
[java] BooleanGetMethodName: False-negatives with `Boolean` wrapper

### ISSUE THREAD

**Issue by phansys** [pmd/pmd#5253]
**Is your feature request related to a problem? Please describe.**
Methods using `Boolean` return declaration are not managed by the `BooleanGetMethodName` rule.

**Describe the solution you'd like**
Update the rule to consider the `Boolean` wrapper too.

**Describe alternatives you've considered**
If this is considered as a BC break, I guess a property enabling the detection for the `Boolean` type could work.

**Additional context**
```java
Boolean getFoo() {
    return true;
}
```

https://github.com/pmd/pmd/blob/a0818d5ab2669dfd7f6bf8ca9e0ec8d8e0148b5d/pmd-java/src/main/resources/category/java/codestyle.xml#L187-L192

**Comment by Aryant-Tripathi** [comment]
Hello @phansys, I would like to work on this issue.  I didn't get clarification on what we need to do here. Could you explain a little about what we need to perform here? I will be happy to do.

**Comment by phansys** [comment]
Hi @Aryant-Tripathi!

>Could you explain a little about what we need to perform here?

The rule `BooleanGetMethodName` is catching cases like this:
```java
boolean getFoo() {
    return true;
}
```
but isn't detecting any violation in cases like this:
```java
Boolean getFoo() {
    return true;
}
```

Please, let me know if you need more context.
Thank you in advance.

**Comment by Aryant-Tripathi** [comment]
@phansys sir, If possible may you be able to mark/tag PR as hacktoberfest-accepted / hacktoberfest label. It will help me in contributing a PR in hacktoberfest.

**Comment by phansys** [comment]
I haven't permission for that @Aryant-Tripathi, sorry.
Maybe @jsotuyod can help.

**Comment by jsotuyod** [comment]
Sure, I've just added the `hacktoberfest` topic to the repo. Happy coding!
